tests/Integration/UI/AdminColumnsParselyStatsTest.php

```diff
@@ -64,6 +65,382 @@ public function tear_down(): void {
 		$this->set_permalink_structure( '' );
 	}
 
+	/**
+	 * Asserts the enqueueing status of the feature's assets according to the
+	 * passed filter values.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param mixed $global_filter_value The value of the global Content Helper filter.
+	 * @param mixed $feature_filter_value The value of the feature filter.
+	 * @param bool  $expected Whether the assets should be enqueued.
+	 */
+	protected function assert_enqueued_status(
+		$global_filter_value, $feature_filter_value, bool $expected
+	): void {
+		parent::set_filters(
+			Admin_Columns_Parsely_Stats::get_feature_filter_name(),
+			$global_filter_value,
+			$feature_filter_value
+		);
+
+		$this->set_valid_conditions_for_parsely_stats();
+		$this->mock_parsely_stats_response( array() );
+		$this->assert_parsely_stats_admin_script( $expected );
+		$this->assert_parsely_stats_admin_styles( $expected );
+	}
+
+	/**
+	 * Verifies that the run() method enqueues the assets by default.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_feature_filter_name
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::get_global_filter_name
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::__construct
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::run
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::should_be_enabled
+	 * @covers \Parsely\RemoteAPI\Remote_API_Base::is_user_allowed_to_make_api_call
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::is_enabled_by_filters
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::set_current_screen
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::is_tracked_as_post_type
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::enqueue_parsely_stats_script_with_data
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_script_id
+	 *
+	 * @group content-helper
+	 * @group content-helper-admin-stats-column
+	 */
+	public function test_assets_get_enqueued_by_default(): void {
+		self::assert_enqueued_status( null, null, true );
```

Comment:
The `test_assets_*` functions specifically test asset loading in relation to the value of global/feature filters. I agree that there is some overlap with existing Parse.ly Stats tests, but this is not exactly what the existing tests are testing (though I'm sure some of them would fail as a side-effect in case of an issue).

We do introduce some extra code to have these new tests, but having more tests and aiming for testing a broader set of values (even if today they are being defaults or assumed to be a certain value) is a good thing in my opinion, especially as it will be clear from the tests where the issue is exactly.

Reply:
My recommendation: Leave this as is.